frontend/src/metabase/containers/SavedQuestionLoader.jsx

```diff
@@ -54,23 +53,16 @@ const SavedQuestionLoader = ({
     await dispatch(loadMetadataForCard(loadedQuestion.card()));
   }, [loadedQuestion?.id()]);
 
-  useEffect(() => {
-    if (loadedQuestion?.id() == null) {
-      return;
-    }
-
+  const question = useMemo(() => {
```

Comment:
The `useMemo` is semantically equivalent to previous `useState` + `useEffect` but it helps avoid a re-render. There was a case where this extra intermediate state (in-between `setQuestion` call and the re-render caused by it) caused things to crash down the tree.